### PR TITLE
fix(VTreeview): prevent redundant treeview re-render during open

### LINKED ISSUE #19919: [Bug Report][3.6.8] VTreeview slow performance on expand
### Environment
**Vuetify Version:** 3.6.8
**Vue Version:** 3.4.27
**Browsers:** Chrome 125.0.0.0
**OS:** Mac OS 10.15.7

### Steps to reproduce
1. Load reproduction link
2. Expand a node

### Expected Behavior
Should be as fast as using VList: 
https://play.vuetifyjs.com/#eNp1k02P2jAQhv/KKKoEaMlCW/WShpVW26qnXlrtCZDWjQ24dWzLH5QK5b93kuDEhHBJ7JlnPt6xvT4n1hSLZ60fj54lWZI7VmpBHHvaSID8mBbEUCgEsXa1ScpTSrxTmwT+cuoOaPm4XG6Shm1owa2DY7pTBn2/gUv4EAEdku6N8hqORHiG4DMtuYwo5EIbmMwK5TJSOH4krsl7Bm2UtlBdhfTJOQZj3C8uKdINix077kRcLF9EeFx6EY0gMkd0XDSIndaOOfBZLbpeY82Yy/6wf4jxgTV0VUc8NpsBEGbUAJzG3kU/1qClmWs4jouxPUnc1EeJm4HA3BaG65ZiJ62MA8p2xAsH5zYRJY5kMJ3B6gmmFxuA0kxmsJ68WmbsZDsPdlLP16Knb3Q9+U4k2bOSSTeZw6SkPCVFobx0aYmFuBYsVd4JLlmUqQ78yZzjcm9DWKH2I2S/Koynw+IvhqHYkEELb+8U+8EIDdiO3+3pVeNEuny+3V0jX5hgPULb3W2/1eyyKFSpvWM0CzOH9g6FsXdWPPUF0hKf2TfmXg5cUMMkrIDTW67OW5NSUWaRWW8Hbry6MMXegKN3+Rl/Obz/VC8eHmY3yaBN9Ki9PfT3oHNiy9j/27szp1WK3+otGknHNHccsa8ngpeQQVFLaMTCnZhqNjBVg71hzhvZNnflisD7gxidwXJkBvfkB+U8an1M6ajGa3lRw2OiqnBtmj/C+aJ7vEk1l16I5rP9DxkApl0=

### Actual Behavior
Slow and frozen

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqVkt1ugzAMhV/FQrsA9YfulgHSNE17iFKpiLhqpBSiYFgllHefgTFSVC52E2P7+MgfybHzalOE71rv2wa9yIsJb1rlhGlWAsTtjgxiK/EbIsmtOsm8IWZeGodzl9Vx6IxyWhdGahps8K4rQyDwkjeKoOtrAEV10w2hiKYCwGAdgR9Aks7VXlrWBF9IH1ephMESEpBiIZpkZSWwZsHx5PYulQFfIYHk1uGNQwyvh/5jswkebWC02OumvvqLDu/IC59fOinsjk973i4FJEkhaz7vOf8PhKLfeUCDZwM2cHPrJgapMeW4zVx3JOvA/8Bdh+1R5WLdJd0q14DNVxW59+bL4EHmwjtcz8Dt7+AURU759FT8zo6+7BGHfy/Ps9uyUWo4Tj9cQ9D5)

### PR DESCRIPTION
**Pull request by yuwu9145**
fixes #19919

## Description

- Open action **SHOULD NOT** trigger any re-rendering
- VTreeview has redundant re-rendering from `v-model:opened` change
- This fix should prevent some of redundant renderings and enhance performance noticeably
- After this PR, the remaining redundant rendering should be resolved in the upstream library babel-plugin-jsx

## Markup:

```vue
<template>
  <v-treeview 
    :items="items"
    @update:opened="openedUpdate"
    ></v-treeview>
</template>

<script>
  export default {
    computed: {
      items: () => {
        const GetChildren = id => {
          const nodes = []
          for (let i = 0; i < 100; i++) {
            nodes.push({
              id: `${id}-${i}`,
              title: `Example child item ${i}`,
            })
          }
          return nodes
        }
        const nodes = []
        for (let i = 0; i < 100; i++) {
          nodes.push({
            id: i,
            title: `Example item ${i}`,
            children: GetChildren(i),
          })
        }
        return nodes
      },
    },
    methods: {
openedUpdate: function(e) {
  console.log('openedUpdate', e)
}
    },
    data: () => ({}),
  }
</script>

```

### CODE REVIEW

File: packages/vuetify/src/labs/VTreeview/VTreeview.tsx [hunk 1]
```diff
@@ -146,7 +147,8 @@ export const VTreeview = genericComponent<new <T>(
     })
 
     useRender(() => {
-      const listProps = VList.filterProps(props)
+      const listProps = VList.filterProps(vm.vnode.props!)
```
Comment (yuwu9145) [inline packages/vuetify/src/labs/VTreeview/VTreeview.tsx:150]: Avoid non-existing prop being assigned as `undefined` to VList

If `@update:opened="openedListener"` is used in Treeview, `VList.filterProps(props)` is equivalent to `<VList :opened="undefined" @update:opened="openedListener">`, which breaks VList [demo](https://play.vuetifyjs.com/playgrounds/Ba6uEA)

**Comment by johnleider** [comment]
This does provide an improvement, but it's still very slow and only appears to work when opening and closing a single group. When you move to the next one or back, it's still a 2-3 second delay before it displays.

**Comment by yuwu9145** [comment]
> This does provide an improvement, but it's still very slow and only appears to work when opening and closing a single group. When you move to the next one or back, it's still a 2-3 second delay before it displays.

The remaining latency is caused by https://github.com/vuejs/babel-plugin-jsx/issues/712. If you remove `v-slots` usages in lines 120, 129 & 138 in VTreeviewChildren.tsx on top of this PR, expanding will be zero latency.

**Review by johnleider — APPROVED** [review]

**Comment by landynfrancis** [comment]
Looks like there has been some progress upstream [here](https://github.com/vuejs/babel-plugin-jsx/issues/712#issuecomment-2781293946)

**Pull request merged**